presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java

```diff
@@ -131,7 +133,15 @@ public FixedSourcePartitionedScheduler(
                     sourceScheduler.noMoreLifespans();
                 }
                 else {
-                    LifespanScheduler lifespanScheduler = new LifespanScheduler(bucketNodeMap, partitionHandles, concurrentLifespansPerTask);
+                    LifespanScheduler lifespanScheduler;
+                    if (!bucketNodeMap.isDynamic()) {
```

Comment:
I'd invert this so that condition is in the positive form (since both paths are very simple).